src/Haskell/Ide/Engine/Options.hs

```diff
@@ -15,6 +29,27 @@ data GlobalOpts = GlobalOpts
   , optFiles         :: [FilePath]
   } deriving (Show)
 
+initApp :: String -> IO GlobalOpts
```

Comment:
Since this is now happening out of context, it would be nice if there was some documentation what this function does, e.g. initialises logger, searches for some project root, etc...

Reply:
Maybe instead reestablish context, such as by saying in a comment that this is the common prefix of hie and hie-wrapper?

Reply:
Yeah, that makes sense, too. Still, what it roughly does to set-up the application would be nice as well.